### ISSUE TITLE
wgpu Implementation (Later)

### ISSUE THREAD

**Issue by VocalFan** [obhq/obliteration#466]
This is mainly so we can give updates and plan on this for later.

As of now, seems Wgpu is getting a performance boost:

https://gfx-rs.github.io/2023/11/24/arcanization.html

**Comment by ultimaweapon** [comment]
We will support multi-backend the same as CPU. The first priority will be Vulkan because it give the best performance so we don't end up rewriting the graphic backend if we hit the limitation of wgpu.

**Comment by VocalFan** [comment]
So... Wgpu's vulkan first, then potentially branch out later.

**Comment by ultimaweapon** [comment]
No, I mean pure Vulkan first while making our code support multiple backends. If someone interested to implement another backend they can work on it the same as @bayedieng currently working on LLVM.

**Comment by VocalFan** [comment]
Ah, fun x3
